assembly/bindgen.ts

```diff
@@ -224,9 +246,9 @@ function decode<T, V = Uint8Array>(buf: V, name: string = ""): T {
   }
   //@ts-ignore
   if (value instanceof u128) {
-    assert(val instanceof JSON.Str);
+    assert(val instanceof JSON.Str, "Value with Key: " + name + " expected type string to decode u128 but got " + JSONTypeToString(val));
     //@ts-ignore
     return u128.fromString(getStr(val, name));
   }
-  unreachable();
+  throw new Error("Bad Decoding.");
```

Comment:
@willemneal I think it would really help to have more detailed message then "Bad Decoding". Bad decoding of what? Why? 

Shouldn't we at least be able to show class name? Something like `Cannot deserialize Foobar from JSON because of unexpected value`